src/ploomber_core/validate.py

```diff
@@ -0,0 +1,111 @@
+import difflib
+from ploomber_core.exceptions import MissingKeysValidationError, ValidationError
+from ploomber_core.io import pretty_print
+import collections.abc
+
+
+def _is_set_or_sequence(obj) -> bool:
+    """
+    Checks if obj is a sequence or a set.
+    Returns False if obj is a string.
+    """
+    if isinstance(obj, str):
+        return False
+    if isinstance(obj, collections.abc.Sequence):
+        return True
+    if isinstance(obj, set):
+        return True
+
+
+def keys(valid, passed, required=None, name='spec', show_matches=True):
+    """
+    Checks if given keys are valid. If not, raises an error with suggested valid keys.
+
+    Parameters
+    ----------
+    valid : list
+        A list of valid values
+
+    passed : str, list, tuple, or a set
+        The value to check
+
+    required : list
+        Required values
+
+    name : str, default 'spec'
+        The name of the parameter
+
+    show_matches : bool, default True
+        Show suggested keys in the error message
+
+    Raises
+    ------
+    ValidationError if passed elements are not in valid
+
+    MissingKeysValidationError if passed elements are not in required
+    """
+    is_set_or_sequence = _is_set_or_sequence(passed)
+
+    if not is_set_or_sequence or passed is None:
+        passed = [passed]
+
+    passed = set(passed)
+
+    if valid:
+        extra = passed - set(valid)
+
+        if extra:
+
+            err_message = f"Error validating '{name}', "
+
+            if len(extra) < 2:
+                err_message += "the following key isn't valid: "
```

Comment:
let's use "value" instead of "key" since this is for function arguments. it'll make it clearer